### PR TITLE
CI implementation,  recording should retry check and moderator check removed

### PR DESCRIPTION
**Pull request by harishnv**
CI implemented
docker file added
moderator check and should retry check removed

### CODE REVIEW

File: .github/workflows/CI-CD.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+name: CI
+on:
+  push:
+    branches:
+      - docker_workflow
+      - develop
+    tags:
+      - v*
```
Comment (manquer) [inline .github/workflows/CI-CD.yml:8]: ?

File: .github/workflows/CI-CD.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+              TAGS="$TAGS,${DOCKER_IMAGE}:sha-${GITHUB_SHA::8}"
+            fi
+            echo ::set-output name=version::${VERSION}
+            echo ::set-output name=tags::${TAGS}
+            echo ::set-output name=created::$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+      - name: Build jar
+        run: mvn install -Dassembly.skipAssembly=false -DskipTests
+        env:
+          WORKSPACE: ${{ github.workspace }}
+          MAVEN_OPTS: "-Xmx4G"
```
Comment (manquer) [inline .github/workflows/CI-CD.yml:68]: CI has 4G ?

File: .github/workflows/CI-CD.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+              MAJOR=${MINOR%.*}
+              TAGS="$TAGS,${DOCKER_IMAGE}:${MINOR},${DOCKER_IMAGE}:${MAJOR},${DOCKER_IMAGE}:latest"
+            elif [ "${{ github.event_name }}" = "push" ]; then
+              TAGS="$TAGS,${DOCKER_IMAGE}:sha-${GITHUB_SHA::8}"
+            fi
+            echo ::set-output name=version::${VERSION}
+            echo ::set-output name=tags::${TAGS}
+            echo ::set-output name=created::$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+      - name: Build jar
+        run: mvn install -Dassembly.skipAssembly=false -DskipTests
```
Comment (manquer) [inline .github/workflows/CI-CD.yml:65]: Dockerfile should contain mvn install

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    ADD . /jicofo
+
+    RUN cd /jicofo
+
+    RUN unzip target/jicofo-1.1-SNAPSHOT-archive.zip -d target/
+    RUN mv /usr/share/jicofo/ /tmp
+    RUN mkdir /usr/share/jicofo/
+    RUN cp -ru target/jicofo-1.1-SNAPSHOT/* /usr/share/jicofo/
+
+    VOLUME /config
```
Comment (manquer) [inline Dockerfile:22]: This is optimized systemd type dameon deploy not docker . 
In docker service in FG

File: rootfs/defaults/jicofo.conf [hunk 5]
```diff
@@ -0,0 +1,159 @@
+{{ $ENABLE_AUTH := .Env.ENABLE_AUTH | default "0" | toBool }}
```
Comment (manquer) [inline rootfs/defaults/jicofo.conf:1]: This is helm template

File: rootfs/defaults/logging.properties [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{{ if .Env.SENTRY_DSN | default "0" | toBool }}
```
Comment (manquer) [inline rootfs/defaults/logging.properties:1]: Helm

File: rootfs/etc/services.d/jicofo/run [hunk 7]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/with-contenv bash
```
Comment (manquer) [inline rootfs/etc/services.d/jicofo/run:1]: This kind of setup not recommended in docker
@Harishkrishna17

File: src/main/java/org/jitsi/jicofo/jibri/JibriSession.java [hunk 8]
```diff
@@ -645,16 +645,16 @@ private void handleJibriStatusUpdate(
...
-            if (shouldRetryParam == null)
-            {
-                logger.warn("failureReason was non-null but shouldRetry wasn't set, will NOT retry");
-                shouldRetry = false;
-            }
-            else
-            {
-                shouldRetry = shouldRetryParam;
-            }
+            boolean shouldRetry=true;
```
Comment (manquer) [inline src/main/java/org/jitsi/jicofo/jibri/JibriSession.java:648]: put a comment a reason why

File: src/main/kotlin/org/jitsi/jicofo/jibri/BaseJibri.kt [hunk 9]
```diff
@@ -161,10 +161,10 @@ abstract class BaseJibri internal constructor(
             return error(iq, StanzaError.Condition.bad_request, "undefined action")
         }
 
-        verifyModeratorRole(iq)?.let {
-            logger.warn("Ignored Jibri request from non-moderator.")
-            return IQ.createErrorResponse(iq, it)
-        }
+        // verifyModeratorRole(iq)?.let {
```
Comment (manquer) [inline src/main/kotlin/org/jitsi/jicofo/jibri/BaseJibri.kt:168]: put a comment explaining why

**Review by manquer — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/jitsi/jicofo/jibri/BaseJibri.kt (hunk 9 above)
Comment (harishnv) [reply src/main/kotlin/org/jitsi/jicofo/jibri/BaseJibri.kt:168]: comment added

File: rootfs/defaults/jicofo.conf (hunk 5 above)
Comment (harishnv) [reply rootfs/defaults/jicofo.conf:1]: This is not a helm template. On boot, this creates jicofo.conf file based on env variable. by using tpl.

File: rootfs/defaults/logging.properties (hunk 6 above)
Comment (harishnv) [reply rootfs/defaults/logging.properties:1]: This is a code template.

File: rootfs/defaults/jicofo.conf (hunk 5 above)
Comment (harishnv) [reply rootfs/defaults/jicofo.conf:1]: https://github.com/talview/jicofo/blob/65d0a3d612d59052af8c467a44818bf4742fd9c4/rootfs/etc/cont-init.d/10-config#L17

**Review by manquer — APPROVED** [review]

**Comment by manquer** [comment]
We will need to move to FG service as entry point and not rely on healthcheck

**Pull request merged**